app/controllers/api/v1/questionnaires_controller.rb

```diff
@@ -0,0 +1,169 @@
+class QuestionnairesController < ApplicationController
+  include AuthorizationHelper
+
+  # Controller for Questionnaire objects
+  # A Questionnaire can be of several types (QuestionnaireType)
+  # Each Questionnaire contains zero or more questions (Question)
+  # Generally a questionnaire is associated with an assignment (Assignment)
+
+  
+  skip_before_action :verify_authenticity_token
+
+  # Check role access for edit questionnaire
+  def action_allowed?
+    case params[:action]
+    when 'edit'
+      @questionnaire = Questionnaire.find(params[:id])
+      current_user_has_admin_privileges? ||
+        (current_user_is_a?('Instructor') && current_user_id?(@questionnaire.try(:instructor_id))) ||
+        (current_user_is_a?('Teaching Assistant') && session[:user].instructor_id == @questionnaire.try(:instructor_id))
+    else
+      current_user_has_student_privileges?
+    end
+  end
+
+  #List all the questionnaires
+  # GET on /questionnaires
+  def index
+    @questionnaires = Questionnaire.order(:id)
+    render json: @questionnaires
+  end
+  
+  #Show a given questionnaire
+  # GET on /questionnaires/:id
+  def show
+    begin
+      @questionnaire = Questionnaire.find(params[:id])
+      render json: @questionnaire
```

Comment:
status required

Reply:
Commit [f039d84](https://github.com/expertiza/reimplementation-back-end/pull/20/commits/f039d843a0176aa490bc542fe0cf6d42cca62d49) addresses this review